### DISCUSSION TITLE
Seeed studio esp32c3

### DISCUSSION THREAD

**Discussion by puppet13th** [micropython/micropython#9146] [General]
Seeed studio esp32c3 look very interesting,very small device with alot of feature.Have anyone got hands-on?
According to [seeed wiki](https://wiki.seeedstudio.com/XIAO_ESP32C3_Getting_Started/) it's only support arduino. Have anyone tried to run MP firmware on it?

**Comment by d-c-d** [comment]
[circuitpython.org](https://circuitpython.org/)  ( yes, you asked about MP ), lists [10 esp32c3](https://circuitpython.org/downloads?q=esp32c3) devices, includeing the [Seeed Studio XIAO ESP32C3](https://circuitpython.org/board/seeed_xiao_esp32c3/)

**Reply by puppet13th** [reply]
Look like circuitpython just added support for seeed esp32c3.

**Comment by jimmo** [comment]
There's no specific board support for that Seeed board, but the generic ESP32 C3 firmware should work. https://micropython.org/download/esp32c3-usb/

I've had a few reports that the BLE is not fully functional yet though, more work required. (Edit: see below, might be OK?)

**Reply by mattytrentini** [reply]
I wonder if we should support adding boards that are simply entries for the download manager but link to generic firmware images?

We've resisted adding boards (particularly ESP32-based) that don't have anything particularly unique about them because it would consume time/space for firmware images that are identical to others. But there is a case to be made that it is helpful to find your specific board in the download manager...

**Reply by puppet13th** [reply]
@jimmo since I switch to aioble library, ble works great on esp32-c3. I have tried it using 1.19.1 firmware on ai-thinker esp32-c3s and lilygo-t-oi board. What function did u mean? BTW,i think seeed esp32c3 should work using USB firmware. It does not have external uart chip/interface.

**Reply by jimmo** [reply]
@mattytrentini Yes it would be very easy to add aliases to an existing board.json (i.e. image+description). It would definitely be good to speed up the build process.

Ideally we add pin naming support to ESP32 though (it's...on the list :) ).

> ble works great on esp32-c3.

OK that's great to know, thanks @puppet13th

**Reply by steveh127** [reply]
I can can confirm the generic ESP32 C3 image works with the Seeed XIAO ESP32 C3 board. WIFI works fine though I haven't tried BLE. I do like the form factor of these tiny boards and the price for that matter. Unless there are special features  or limitations then having a generic image is better than a plethora of near identical board images.

**Reply by mattytrentini** [reply]
@steveh127 the problem is that many people don't realise which image they should download for their board. And it's common for boards to have different ways to deploy the firmware. So having a board-specific `deploy.md`/`board.json` but linking to a generic image would have advantages.

**Reply by puppet13th** [reply]
@steveh127 which firmware did u flashed? no-usb/USB one? I'm soo eager to try this tiny device. Last time I checked at their website, still got back order until next month. BTW where did u got yours?

**Comment by mattytrentini** [comment]
I [mentioned](https://discord.com/channels/862602258452578314/963715835233439745/1014129107061178458) on the [Seeed Studio Discord](https://discord.gg/QqMgVwHT3X) ([xiao-series](https://discord.com/channels/862602258452578314/963715835233439745) channel) that MicroPython and CircuitPython work well on the XIAO ESP32C3 and it would be helpful to update their wiki. 

Does anyone know of a better place to make that request?

**Reply by mattytrentini** [reply]
Seems like [Seeed are on the case](https://discord.com/channels/862602258452578314/963715835233439745/1017627738674315404): "...MicroPython is now in the test environment, and we will release it once it is verified"
